docs/content/testing/test-guidelines.md

```diff
@@ -0,0 +1,241 @@
+ the first parameter is the expected value, and the second parameter is the
+ value we are testing. (For an example, see the code snippet below.)
+
+```cs
+  var sumResult = Add(2, 3);
+  Assert.AreEqual(5, sumResult);
+```
+
+#### Assertions for exceptions
+
+By default, tests will fail if any code within it throws an exception.
+
+Sometimes, you may want to test for conditions which throw exceptions. You can do this using the `Assert.Throws` method. This method returns the caught exception, and you can test further assertions against that. (For an example, see the code snippet below.)
+```cs
+        [Test]
+        public void Validate_should_return_false_when_LocatorHost_is_empty()
+        {
+            var config = GetDefaultWorkingConfig();
+            config.LocatorHost = "";
+
+            var exception = Assert.Throws<System.ArgumentException>(config.Validate);
+            Assert.IsTrue(exception.Message.Contains("locatorHost"));
+        }
+```
+
+### Disposables vs setup and teardown
+
+If you have a fixture that has only one test and the setup is simple, use the `using` keyword within the test. (For an example, see the code snippet below.)
+
+
+```cs
+[Test]
+public void World_should_have_an_EntityManager()
+{
+    using (var world = new World("test-world"))
+    {
+        Assert.IsNotNull(world.GetOrCreateManager<EntityManager>());
+    }
+}
+```
+
+However, if you are going to  add a similar test to the same
+ fixture (for example, you need to create another `World` to assert something else about
+ the `World`) move the creation and disposal of the instance into the `[SetUp]`
+ and `[TearDown]` functions, as in the code snippet below.
+
+```cs
+private World world;
+
+[SetUp]
+public void SetUp()
+{
+    world = new World("test-world")
+}
+
+[TearDown]
+public void TearDown()
+{
+    world.Dispose();
+}
+
+[Test]
+public void World_should_have_an_EntityManager()
+{
+    Assert.IsNotNull(world.GetOrCreateManager<EntityManager>());
+}
+
+[Test]
+public void World_should_not_have_behaviour_managers_when_created()
+{
+    Assert.AreEqual(0, world.BehaviourManagers.Count());
+}
+```
+
+### Handling Unity logs
+
+By default, any error logs will fail your test.
+
+You can choose to expect these errors or ignore them using the `LogAssert`
+ class. Please refer to the [Unity  documentation](https://docs.unity3d.com/ScriptReference/TestTools.LogAssert.html)
+ for use details.
+
+We recommend that you:
+* Expect any error logging that the modules of the Spatialos GDK for Unity outputs.
+* Expect any error logging that comes from other parts of the Unity engine, or third parties that are relevant to the functionality that you are testing.
+* Ignore irrelevant error logs as last resort.
+
+Do not forget to reset `LogAssert.ignoreFailingMessages` to false every time you modify it for a test.
+
+
+[//]: # (Editorial review status: Full review 2018-07-13)
+[//]: # (Questions to deal with (but not limited to):)
```

Comment:
This says nothing and should be deleted

Reply:
Syntax part of the review status via https://docs.google.com/document/d/1iiozdhFcpZkTWLKdRqA8Jgef0zrZpFWRJuD1e1KTevY/edit